### ISSUE TITLE
Bug Report: Rank doesn't save on custom variations.

### ISSUE THREAD

**Issue by kidwithagun** [FunkinCrew/Funkin#3102]
### Issue Checklist

- [X] I have properly named the issue
- [X] I have checked the issues/discussions pages to see if the issue has been previously reported

### What platform are you using?

Itch.io (Downloadable Build) - Windows

### If you are playing on a browser, which one are you using?

None

### Version

0.4.1

### Context (Provide images, videos, etc.)

After beating a song with a custom variation, hovering over another song and then hovering back will not display the rank you got but will keep the score for some reason.

Here I just added another variation to fresh as an example for what happens: (sorry for the poor quality btw, i dont have a screen recorder)

https://github.com/user-attachments/assets/4cdbb88b-9a32-449a-9cdb-06ddd53706bb

### Steps to reproduce (or crash logs, errors, etc.)

Add a variation to a song.

Beat the difficulty given to the variation.

Hover over another song then hover back and the rank is gone.

**Comment by PortilizenDev** [comment]
Why was the song name text squished like that

**Comment by Lasercar** [comment]
Does this still happen in 0.5.3?

**Comment by Hundrec** [comment]
Seeing as remix mods like FNF: Polished and Spooky Mix actually work, I'm inclined to say this has been fixed.
If it pops up again, someone can create a new issue!

**Comment by Hundrec** [comment]
That's true, so I'm willing to accommodate if someone reports this issue again

**Comment by AbnormalPoof** [comment]
> > Seeing as remix mods like FNF: Polished and Spooky Mix actually work, I'm inclined to say this has been fixed. If it pops up again, someone can create a new issue!
> 
> I mean those are source code mods so they might have fixed it by themselves

Spooky Mix has a softcoded version.

This no longer occurs.

![Image](https://github.com/user-attachments/assets/7f685ff3-e940-412a-8ec6-ed4a712ead8d)

**Comment by sphis-Sinco** [comment]
> > > Seeing as remix mods like FNF: Polished and Spooky Mix actually work, I'm inclined to say this has been fixed. If it pops up again, someone can create a new issue!
> > 
> > 
> > I mean those are source code mods so they might have fixed it by themselves
> 
> Spooky Mix has a softcoded version.
> 
> This no longer occurs.
> 
> ![Image](https://github.com/user-attachments/assets/7f685ff3-e940-412a-8ec6-ed4a712ead8d)

oh hey, they made it
